integration_tests/web/test_conversations.py

```diff
@@ -0,0 +1,129 @@
+from slack_sdk.web.async_client import AsyncWebClient
+
+
+class TestWebClient(unittest.TestCase):
+    """Runs integration tests with Slack API for conversations.* endpoints
+    To run, we use two workspace-level bot tokens,
+    one for the inviting workspace(list and send invites) another for the recipient
+    workspace (accept and approve) sent invites. Before being able to run this test suite,
+    we also need to have manually created a slack connect shared channel and added
+    these two bots as members first. See: https://api.slack.com/apis/connect
+
+    In addition to conversations.connect:* scopes, your sender bot token should have channels:manage scopes.
+    """
+
+    def setUp(self):
+        self.logger = logging.getLogger(__name__)
+        self.sender_bot_token = os.environ[SLACK_SDK_TEST_SENDER_BOT_TOKEN]
+        self.receiver_bot_token = os.environ[SLACK_SDK_TEST_RECEIVER_BOT_TOKEN]
+        self.sender_sync_client: WebClient = WebClient(token=self.sender_bot_token)
+        self.sender_async_client: AsyncWebClient = AsyncWebClient(
+            token=self.sender_bot_token
+        )
+        self.receiver_sync_client: WebClient = WebClient(token=self.receiver_bot_token)
+        self.receiver_async_client: AsyncWebClient = AsyncWebClient(
+            token=self.receiver_bot_token
+        )
+
+    def tearDown(self):
+        pass
+
+    def test_sync(self):
+        sender = self.sender_sync_client
+        receiver = self.receiver_sync_client
+
+        # list senders pending connect invites
+        connect_invites: SlackResponse = sender.conversations_listConnectInvites()
+        self.assertIsNotNone(connect_invites["invites"])
+
+        # creates channel in sender workspace to share
+        unique_channel_name = str(int(time.time())) + "-shared"
+        new_channel: SlackResponse = sender.conversations_create(
+            name=unique_channel_name
+        )
+        self.assertIsNotNone(new_channel["channel"])
+        self.assertIsNotNone(new_channel["channel"]["id"])
+
+        # send an invite for sender's intended shared channel to receiver's bot user id
+        invite: SlackResponse = sender.conversations_inviteShared(
+            channel=new_channel["channel"]["id"],
+            user_ids=os.environ[SLACK_SDK_TEST_RECEIVER_BOT_USER_ID],
+        )
+        self.assertIsNotNone(invite["invite_id"])
+
+        # reciever accept conversations invite via invite id
+        accepted: SlackResponse = receiver.conversations_acceptSharedInvite(
+            channel_name=unique_channel_name,
+            invite_id=invite["invite_id"],
+        )
+        self.assertIsNone(accepted["error"])
+
+        # receiver attempt to approve invite already accepted by an admin level token should fail
+        self.assertRaises(
+            SlackApiError,
+            receiver.conversations_approveSharedInvite,
+            invite_id=invite["invite_id"],
+        )
+
+        # clean up created channel
```

Comment:
How about having this part with `if channel_id is not None:` in `finally` clause? We can have a new local variable `channel_id: Optional[str]` outside the try/finally block.

Reply:
Yes, that is much nicer - and we will always attempt to cleanup any channels in the course of the test. Thanks for the suggestion - I've included these changes!